### PR TITLE
[FIX] point_of_sale: load partners based on email and name with search

### PR DESCRIPTION
**Pull request by pedrambiria**
Before this commit: if the limited partner load option was enabled, when a
user tried to load a user by searching, it would only search based on the
name. In several cases, it's necessary to search based on the email
like when you are searching in the contact in the backend.

The solution is to add other fields to the domain of search.

opw-2952814

X-original-commit: dba0eca41f2922a9054c5c926b24d7f570382bd7

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerListScreen.js [hunk 1]
```diff
@@ -184,8 +184,11 @@ odoo.define('point_of_sale.PartnerListScreen', function(require) {
         async getNewPartners() {
             let domain = [];
             if(this.state.query) {
-                domain = ['|', ["name", "ilike", this.state.query + "%"],
-                               ["parent_name", "ilike", this.state.query + "%"]];
+                domain = ['|', '|',
+                    ["parent_name", "ilike", this.state.query],
```
Comment (pimodoo) [inline addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerListScreen.js:188]: Why did you remove + "%" ?

Comment (pedrambiria) [reply addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerListScreen.js:188]: In the domain it's not necessary.

**Comment by pedrambiria** [comment]
It has been fixed
